src/DegenGambit.sol

```diff
@@ -272,12 +267,36 @@ contract DegenGambit is ERC20, ReentrancyGuard {
         address player,
         uint256 amount,
         uint256 prizeIndex
-    ) internal {
-        winners[prizeIndex] = Winner({
-            player: player,
-            amount: amount,
-            timestamp: block.timestamp
-        });
+    ) internal virtual {
+        if (prizeIndex == 0) {
```

Comment:
MAybe rename to `_updateWinners` to reflect that it's internal.

```
require(prizeIndex <= 6, "Invalid prizeIndex");
```

or something like this?